TreeMaker/python/maker.py

```diff
@@ -103,6 +106,31 @@ def __init__(self,parameters):
 
         self.readFiles = [(self.redir if val.startswith("/") else "")+val for val in self.readFiles]
 
+        # https://htcondor.readthedocs.io/en/latest/man-pages/condor_chirp.html?highlight=set_job_attr_delayed#chirp-commands
+        # https://github.com/cms-sw/cmssw/blob/master/FWCore/Services/plugins/CondorStatusUpdater.cc#L377-L410
+        try:
+            if os.environ["_CONDOR_CHIRP_CONFIG"]:

```

Comment:
the outer try/catch seems unnecessary. why not just:
```python
chirp_env = "_CONDOR_CHIRP_CONFIG"
if chirp_env in os.environ and os.environ[chirp_env]:
    ...
```

Reply:
We check for the environment variable `_CONDOR_CHIRP_CONFIG` to see if we're on a condor worker node. However, when os.environ doesn't find an environment variable it returns a `KeyError`. We don't want that to stop the whole program, we just don't want to add a condor classad in that case. So the outer try catches that error, which we know will be coming from the one possible place which could generate a `KeyError`.

One solution I can come up with is to use:
```python
if os.environ.get("_CONDOR_CHIRP_CONFIG") is not None:
    <other stuff>
```
Would you like that better?

Reply:
the point is that os.environ is just a dict. `if x in dict` doesn't throw an error. since `and` short-circuits, `os.environ[chirp_env]` will only be called if the key is present.

Reply:
Changed